fdbserver/Knobs.cpp

```diff
@@ -371,8 +371,11 @@ ServerKnobs::ServerKnobs(bool randomize, ClientKnobs* clientKnobs) {
 	init( TARGET_BYTES_PER_STORAGE_SERVER_BATCH,               500e6 ); if( smallStorageTarget ) TARGET_BYTES_PER_STORAGE_SERVER_BATCH = 1500e3;
 	init( SPRING_BYTES_STORAGE_SERVER_BATCH,                    50e6 ); if( smallStorageTarget ) SPRING_BYTES_STORAGE_SERVER_BATCH = 150e3;
 	init( STORAGE_HARD_LIMIT_BYTES,                           1500e6 ); if( smallStorageTarget ) STORAGE_HARD_LIMIT_BYTES = 4500e3;
-	init( STORAGE_DURABILITY_LAG_SOFT_MAX,                      20e6 ); if( smallStorageTarget ) STORAGE_DURABILITY_LAG_SOFT_MAX = 10e6;
 	init( STORAGE_DURABILITY_LAG_HARD_MAX,                     200e6 ); if( smallStorageTarget ) STORAGE_DURABILITY_LAG_HARD_MAX = 100e6;
+	init( TARGET_VERSIONS_PER_STORAGE_SERVER,                   20e6 ); if( smallStorageTarget ) TARGET_VERSIONS_PER_STORAGE_SERVER = 10e6;
+	init( SPRING_VERSIONS_STORAGE_SERVER,                        2e6 ); if( smallStorageTarget ) SPRING_VERSIONS_STORAGE_SERVER = 1e6;
+	init( TARGET_VERSIONS_PER_STORAGE_SERVER_BATCH,             10e6 ); if( smallStorageTarget ) TARGET_VERSIONS_PER_STORAGE_SERVER_BATCH = 5e6;
+	init( SPRING_VERSIONS_STORAGE_SERVER_BATCH,                  1e6 ); if( smallStorageTarget ) SPRING_VERSIONS_STORAGE_SERVER_BATCH = 5e5;
 
```

Comment:
Would changing the name to something like  'TARGET_VERSIONS_LAG_PER_STORAGE_SERVER' be better?

Reply:
That might be ambiguous with the other concept that exists to track lag between log and storage server. If you want to include lag in the name, it should probably be something like 'durability_lag', although I think I'm personally ok with the current name being thought of as saying "this is the target number of versions that a storage server should hold in memory".